### PR TITLE
Rdcc 1922

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

RDCC-1922

### Change description ###

feat: LRD Api Get endpoint code is completed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by AdilOoze — APPROVED** [review]

File: charts/rd-location-ref-api/values.yaml [hunk 1]
```diff
@@ -3,16 +3,18 @@ java:
   image: 'hmctspublic.azurecr.io/rd/rd-location-ref-api:latest'
   ingressHost: rd-location-ref-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
   aadIdentityName: rd
-  readinessPath: '/health/readiness'
   environment:
     POSTGRES_HOST: rd-location-ref-api-postgres-db-{{ .Values.global.environment }}.postgres.database.azure.com
-    POSTGRES_USERNAME: dbrdlocationref@rd-location-ref-api-postgres-db-{{ .Values.global.environment }}
-    POSTGRES_DB_NAME: dbrdlocationref
+    POSTGRES_USERNAME: locrefdata@rd-location-ref-api-postgres-db-{{ .Values.global.environment }}
```
Comment (abhijitdiwan) [inline charts/rd-location-ref-api/values.yaml:8]: we changed this dbrdlocationref..it will fail load

File: infrastructure/main.tf [hunk 2]
```diff
@@ -21,8 +21,8 @@ module "db-rd-location-ref-api" {
   location            = var.location
   subscription        = var.subscription
   env                 = var.env
-  postgresql_user     = "dbrdlocationref"
-  database_name       = "dbrdlocationref"
+  postgresql_user     = "locrefdata"
+  database_name       = "locrefdata"
```
Comment (abhijitdiwan) [inline infrastructure/main.tf:25]: agin it should both sync..just doe't change db else it fails load.

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                    message = "Internal Server Error"
+            )
+    })
+    @GetMapping(produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> retrieveOrgServiceDetailsByServiceCodeOrCcdCaseType(
+        @RequestParam(value = "serviceCode",required = false) String serviceCode,
+        @RequestParam(value = "ccdCaseType",required = false) String ccdCaseType) {
+        log.info("inside retrieveServiceCodeDetailsByServiceCodeOrCcdCaseType");
+        List<LrdOrgInfoServiceResponse> lrdOrgInfoServiceResponse = null;
+        if (!StringUtils.isBlank(serviceCode) && !StringUtils.isBlank(ccdCaseType)) {
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java:72]: isNotBank is there.

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/WelcomeController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        )
+    })
+    @GetMapping(
+        path = "/",
+        produces = APPLICATION_JSON_VALUE
+    )
+    @ResponseBody
+    public ResponseEntity<String> welcome() {
+
+        log.info("{}:: Welcome message '{}' from running instance: {}", loggingComponentName, MESSAGE, INSTANCE_ID);
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/WelcomeController.java:60]: what is {} 3rd one here? is it repalcement?

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/InvalidRequestException.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.lrdapi.controllers.advice;
+
+public class InvalidRequestException extends RuntimeException {
+
+    public InvalidRequestException(String message) {
+        super(message);
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/InvalidRequestException.java:6]: why are we creating so many exception classes? I think one should fine LrdApiException

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/LogAndSuppressRequestRejectedExceptionFilter.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@Component
+@Slf4j
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class LogAndSuppressRequestRejectedExceptionFilter extends GenericFilterBean {
+
+    private Gson gson = new Gson();
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/LogAndSuppressRequestRejectedExceptionFilter.java:31]: can't we Autowire this?

File: src/main/java/uk/gov/hmcts/reform/lrdapi/domain/Service.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+@Entity
+@NoArgsConstructor
+@EntityListeners(AuditingEntityListener.class)
+@Getter
+@Setter
+@SequenceGenerator(name = "service_seq", sequenceName = "service_seq", allocationSize = 1)
+@NamedEntityGraph(
+        name = "Service.alljoins",
+        attributeNodes = {
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/lrdapi/domain/Service.java:38]: are we using entity graph? I mean  do we fetching all records?

File: src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/LrdServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            services.forEach(service -> {
+                orgInfoServiceResponses.add(new LrdOrgInfoServiceResponse(service));
+            });
+
+        }
+        return orgInfoServiceResponses;
+    }
+
+    public void ifServiceResponseNullThrowException(Service service) {
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/LrdServiceImpl.java:66]: avoid hard cod null check use utility methods.

File: infrastructure/main.tf (hunk 2 above)
Comment (KOTLAPRA23) [reply infrastructure/main.tf:25]: ok. I will change

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java (hunk 3 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdApiController.java:72]: ok. i will change

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/WelcomeController.java (hunk 4 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/WelcomeController.java:60]: yes .

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/InvalidRequestException.java (hunk 5 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/InvalidRequestException.java:6]: This one is used for 400 error code scenario and LrdApiException is used for internal server error scenario's

File: src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/LrdServiceImpl.java (hunk 8 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/LrdServiceImpl.java:66]: it is fine .

File: charts/rd-location-ref-api/values.yaml (hunk 1 above)
Comment (KOTLAPRA23) [reply charts/rd-location-ref-api/values.yaml:8]: ok . i will change

**Comment by KOTLAPRA23** [comment]
implemented review comments and closing the PR since modifying the existing flyway scripts  
 the new PR will be the below one for your reference

https://github.com/hmcts/rd-location-ref-api/pull/80

File: src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/LogAndSuppressRequestRejectedExceptionFilter.java (hunk 6 above)
Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/LogAndSuppressRequestRejectedExceptionFilter.java:31]: Not sure about @autowire.. this class copied from PRD and it's filter class not sure how to use autowire here?

Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/lrdapi/controllers/advice/LogAndSuppressRequestRejectedExceptionFilter.java:31]: Removed the class which is not required here.

**Pull request closed without merging**